pkg/cloud/packet/client.go

```diff
@@ -225,35 +243,40 @@ func (p *Client) GetDeviceByTags(project string, tags []string) (*packngo.Device
 
 // CreateIP reserves an IP via Packet API. The request fails straight if no IP are available for the specified project.
 // This prevent the cluster to become ready.
-func (p *Client) CreateIP(namespace, clusterName, projectID, facility, metro string) (net.IP, error) {
-	req := packngo.IPReservationRequest{
-		Type:                   packngo.PublicIPv4,
+func (p *Client) CreateIP(ctx context.Context, namespace, clusterName, projectID, facility, metro string) (net.IP, error) {
+	failOnApprovalRequired := true
+	req := metal.IPReservationRequestInput{
+		Type:                   "public_ipv4",
 		Quantity:               1,
 		Facility:               &facility,
 		Metro:                  &metro,
-		FailOnApprovalRequired: true,
+		FailOnApprovalRequired: &failOnApprovalRequired,
 		Tags:                   []string{generateElasticIPIdentifier(clusterName)},
 	}
 
-	r, resp, err := p.ProjectIPs.Request(projectID, &req)
+	apiRequest := p.IPAddressesApi.RequestIPReservation(ctx, projectID)
+	r, resp, err := apiRequest.RequestIPReservationRequest(metal.RequestIPReservationRequest{
+		IPReservationRequestInput: &req,
+	}).Execute() //nolint:bodyclose
 	if err != nil {
 		return nil, err
 	}
 	if resp.StatusCode == http.StatusUnprocessableEntity {
 		return nil, ErrElasticIPQuotaExceeded
 	}
 
-	ip := net.ParseIP(r.Address)
+	rawIP := r.IPReservation.GetAddress()
+	ip := net.ParseIP(rawIP)
 	if ip == nil {
-		return nil, fmt.Errorf("failed to parse IP: %s, %w", r.Address, ErrInvalidIP)
+		return nil, fmt.Errorf("failed to parse IP: %s, %w", rawIP, ErrInvalidIP)
 	}
 	return ip, nil
 }
 
 // enableBGP enable bgp on the project
-func (p *Client) EnableProjectBGP(projectID string) error {
+func (p *Client) EnableProjectBGP(ctx context.Context, projectID string) error {
 	// first check if it is enabled before trying to create it
-	bgpConfig, _, err := p.BGPConfig.Get(projectID, &packngo.GetOptions{})
+	bgpConfig, _, err := p.BGPApi.FindBgpConfigByProject(ctx, projectID).Execute() //nolint:bodyclose
```

Comment:
@ctreatma Can you leave a comment here about why we need the nolint:bodyclose tag?